sound/soc/sof/intel/hda.c

```diff
@@ -53,67 +109,58 @@ static int sdw_clock_stop_quirks = SDW_INTEL_CLK_STOP_BUS_RESET;
 module_param(sdw_clock_stop_quirks, int, 0444);
 MODULE_PARM_DESC(sdw_clock_stop_quirks, "SOF SoundWire clock stop quirks");
 
+static int sdw_dai_config_ipc(struct snd_sof_dev *sdev, struct snd_soc_dai *d,
+			      int link_id, int alh_stream_id, bool setup, int dir)
+{
+	struct sof_ipc_dai_config *config;
+	struct snd_sof_dai *sof_dai;
+	bool found = false;
+
+	/* find snd_sof_dai matching the BE CPU DAI name */
+	list_for_each_entry(sof_dai, &sdev->dai_list, list)
+		if (sof_dai->cpu_dai_name && !strcmp(d->name, sof_dai->cpu_dai_name)) {
+			found = true;
+			break;
+		}
+
+	if (!found) {
+		dev_err(sdev->dev, "error: failed to find SOF DAI for: %s\n", d->name);
+		return -EINVAL;
+	}
+
+	config = sof_dai->dai_config;
+	if (!config) {
+		dev_err(sdev->dev, "error: no config for DAI %s\n", sof_dai->name);
+		return -EINVAL;
+	}
+
+	/* update config with link and stream ID */
+	config->dai_index = (link_id << 8) | d->id;
+	config->alh.stream_id = alh_stream_id;
+
+	return hda_dai_widget_setup(sdev, sof_dai, config, setup, dir);
```

Comment:
we've got to find a better naming here.

It's not intuitive at all you use an 'hda_dai_widget' function to configure a SoundWire DAI which uses a different DMA and different ALH IP.

Reply:
if I move it to sof-audio.c and call is just sof_dai_widget_update(), would that be OK?

Reply:
Ah, I get the problem now. hda can be a link or can be a controller (handled in hda.c).
I think you are using "hda" as a namespace prefix here for the controller, it has nothing to do with hdaudio link/dai.

rename it as hda_ctlr_dai_widget_setup?